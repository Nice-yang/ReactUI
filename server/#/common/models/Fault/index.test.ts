import { Fault } from './';

it('should show stack trace on Error', () => {
  const fault = new Fault('error!!!', { meta: 'foobar' }, 1);

  expect(fault).toMatchSnapshot();

  expect(fault instanceof Fault).toBeTruthy();
});
