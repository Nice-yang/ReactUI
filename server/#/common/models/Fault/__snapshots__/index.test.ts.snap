// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show stack trace on Error 1`] = `
Fault {
  "data": Object {
    "meta": "foobar",
  },
  "level": 1,
  "message": "error!!!",
  "stack": "Error
    at new Fault (/Users/I0459/scratch/personal/beragi/server/#/common/models/Fault/index.ts:11:18)
    at Object.<anonymous>.it (/Users/I0459/scratch/personal/beragi/server/#/common/models/Fault/index.test.ts:4:17)
    at Object.asyncJestTest (/Users/I0459/scratch/personal/beragi/server/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:102:37)
    at resolve (/Users/I0459/scratch/personal/beragi/server/node_modules/jest-jasmine2/build/queueRunner.js:43:12)
    at new Promise (<anonymous>)
    at mapper (/Users/I0459/scratch/personal/beragi/server/node_modules/jest-jasmine2/build/queueRunner.js:26:19)
    at promise.then (/Users/I0459/scratch/personal/beragi/server/node_modules/jest-jasmine2/build/queueRunner.js:73:41)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:189:7)",
}
`;
